### PR TITLE
Wire Mobius loop with overdue search screen UI

### PR DESCRIPTION
**Pull request by janhavisinghh**
https://app.shortcut.com/simpledotorg/story/8531/wire-mobius-loop-with-overdue-screen-ui

### CODE REVIEW

**Comment by janhavisinghh** [comment]
Current dependencies on/for this PR:
* master
  * **PR #4018** <a href="https://app.graphite.dev/github/pr/simpledotorg/simple-android/4018" target="_blank">Graphite</a> 
    * **PR #4019** <a href="https://app.graphite.dev/github/pr/simpledotorg/simple-android/4019" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/simpledotorg/simple-android/4019?utm_source=stack-comment).

File: app/src/main/java/org/simple/clinic/home/overdue/search/OverdueSearchScreen.kt [hunk 1]
```diff
@@ -47,6 +67,12 @@ class OverdueSearchScreen : BaseScreen<
 
   private val disposable = CompositeDisposable()
```
Comment (msasikanth) [inline app/src/main/java/org/simple/clinic/home/overdue/search/OverdueSearchScreen.kt:68]: <i class='graphite__hidden'>[Re: lines 56 to 68]<br/></i>

The placing of these variable seems odd as it breaks the rest of `Inject` variables. Let's move them to `searchQueryTextChanges` variable? <span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/simpledotorg/simple-android/4019?utm_source=unchanged-line-comment">Graphite</a>.</span>

**Review by msasikanth — CHANGES_REQUESTED** [review]
See inline comments

File: app/src/main/java/org/simple/clinic/home/overdue/search/OverdueSearchScreen.kt [hunk 2]
```diff
@@ -116,6 +196,40 @@ class OverdueSearchScreen : BaseScreen<
...
+    overdueSearchHistoryContainer.visibility = View.VISIBLE
+    searchHistoryAdapter.clear()
+    searchHistoryAdapter.addAll(searchHistory)
+  }
+
+  override fun hideSearchResults() {
+    overdueSearchRecyclerView.visibility = View.GONE
+  }
+
+  override fun showOverdueSearchResults(overdueSearchResults: PagingData<OverdueAppointment>) {
```
Comment (msasikanth) [inline app/src/main/java/org/simple/clinic/home/overdue/search/OverdueSearchScreen.kt:209]: The param name is different from what's defined in the `UiActions` interface. So, there is a lint warning. Let's update the name to resolve it.

**Review by msasikanth — APPROVED** [review]

**Pull request merged**